### ISSUE TITLE
How to support binary stl to g-code conversion?

### ISSUE THREAD

**Issue by aksingh108** [dronus/Schlizzer#1]
Hi ,

This Schlizzer code only supports ASCII stl to g-code conversion. 
Is there a way to support binary STL to g-code conversion?

Thanks & regards,
Aks.

**Comment by dronus** [comment]
Yes, it could be done by checking the format inside of `loadStl()` and add another function `loadBinaryStl()` that would parse the binary format. Code left at `loadStl` should the lines  from line 267 down to fill the `triangles` collection in some ordered fashion. 
So the ASCII STL loading could be moved out to another function `loadAsciiStl()` which, like `loadBinaryStl()`, would fill `uniqueVertices`, `indices` and `normals`, to be merged into `triangles` by the existing code at line 267.

**Comment by dronus** [comment]
Both functions `loadAsciiStl` and `loadBinaryStl` would then be called by references to the empty lists `uniqueVertices`, `indices` and `normals` to fill them and the current file handle `file`.

Structure of the binary format can be found at https://en.wikipedia.org/wiki/STL_(file_format) .

**Comment by aksingh108** [comment]
Hi,

Thanks for the prompt response. Could you please tell me how to parse binary format?

With regards,
Aks.

**Comment by dronus** [comment]
No. I can give you some hints, found by googling:
- The file format is described by https://en.wikipedia.org/wiki/STL_(file_format) as already mentioned
- A binary STL reader coded in C++ can be found here: https://github.com/obviousjim/ofxSTL/blob/master/src/utils/ofxSTLBinaryReader.h but you need to adapt it to fill Schlizzer data structures as mentioned above (`uniqueVertices`,`indices`,`normals`)
- A function to automatically decide between ASCII or binary STL is found in the same project, see here from line 53: https://github.com/obviousjim/ofxSTL/blob/master/src/ofxSTLImporter.cpp

**Comment by sjhalayka** [comment]
I added a fork with the binary STL reader code.
